core/typedefs.h

```diff
@@ -124,14 +124,22 @@ constexpr const T SIGN(const T m_v) {
 	return m_v > 0 ? +1.0f : (m_v < 0 ? -1.0f : 0.0f);
 }
 
-template <typename T, typename T2>
-constexpr auto MIN(const T m_a, const T2 m_b) {
-	return m_a < m_b ? m_a : m_b;
+template <typename T, typename T2, typename... Rest>
+constexpr auto MIN(const T m_a, const T2 m_b, const Rest... m_others) {
+	if constexpr (!sizeof...(m_others)) {
+		return m_a < m_b ? m_a : m_b;
+	} else {
+		return MIN(MIN(m_a, m_b), m_others...);
+	}
 }
```

Comment:
I think I would prefer this form:

```suggestion
template <typename T, typename T2>
constexpr auto MIN(const T m_a, const T2 m_b) {
		return m_a < m_b ? m_a : m_b;
}

template <typename T, typename T2, typename T3, typename... Rest>
constexpr auto MIN(const T m_a, const T2 m_b, const T3 m_c, const Rest... m_others) {
		return MIN(MIN(m_a, m_b), m_c, m_others...);
}
```

Perhaps even (not sure if that compilers correctly, I'd have to test):
```c++
template <typename T, typename T2>
constexpr auto MIN(const T m_a, const T2 m_b) {
		return m_a < m_b ? m_a : m_b;
}

template <typename T, typename T2, typename... Rest>
constexpr auto MIN(const T m_a, const T2 m_b, const Rest... m_others) {
		return MIN(MIN(m_a, m_b), m_others...);
}
```

The reason being that the original `MIN` / `MAX` are left intact and can be specialized (if needed), while the varargs template only exists to simplify calls.

Reply:
Good idea, will do that